client/settings/deposits/index.js

```diff
@@ -36,40 +37,14 @@ const daysOfWeek = [
 	{ label: __( 'Friday', 'woocommerce-payments' ), value: 'friday' },
 ];
 
+// Monthly deposit schedule anchors: 1-28 labelled with ordinal suffix and 31 labelled as "Last day of the month".
 const monthlyAnchors = [
-	{ label: __( '1st', 'woocommerce-payments' ), value: 1 },
-	{ label: __( '2nd', 'woocommerce-payments' ), value: 2 },
-	{ label: __( '3rd', 'woocommerce-payments' ), value: 3 },
-	{ label: __( '4th', 'woocommerce-payments' ), value: 4 },
-	{ label: __( '5th', 'woocommerce-payments' ), value: 5 },
-	{ label: __( '6th', 'woocommerce-payments' ), value: 6 },
-	{ label: __( '7th', 'woocommerce-payments' ), value: 7 },
-	{ label: __( '8th', 'woocommerce-payments' ), value: 8 },
-	{ label: __( '9th', 'woocommerce-payments' ), value: 9 },
-	{ label: __( '10th', 'woocommerce-payments' ), value: 10 },
-	{ label: __( '11th', 'woocommerce-payments' ), value: 11 },
-	{ label: __( '12th', 'woocommerce-payments' ), value: 12 },
-	{ label: __( '13th', 'woocommerce-payments' ), value: 13 },
-	{ label: __( '14th', 'woocommerce-payments' ), value: 14 },
-	{ label: __( '15th', 'woocommerce-payments' ), value: 15 },
-	{ label: __( '16th', 'woocommerce-payments' ), value: 16 },
-	{ label: __( '17th', 'woocommerce-payments' ), value: 17 },
-	{ label: __( '18th', 'woocommerce-payments' ), value: 18 },
-	{ label: __( '19th', 'woocommerce-payments' ), value: 19 },
-	{ label: __( '20th', 'woocommerce-payments' ), value: 20 },
-	{ label: __( '21st', 'woocommerce-payments' ), value: 21 },
-	{ label: __( '22nd', 'woocommerce-payments' ), value: 22 },
-	{ label: __( '23rd', 'woocommerce-payments' ), value: 23 },
-	{ label: __( '24th', 'woocommerce-payments' ), value: 24 },
-	{ label: __( '25th', 'woocommerce-payments' ), value: 25 },
-	{ label: __( '26th', 'woocommerce-payments' ), value: 26 },
-	{ label: __( '27th', 'woocommerce-payments' ), value: 27 },
-	{ label: __( '28th', 'woocommerce-payments' ), value: 28 },
-	{
-		label: __( 'Last day of the month', 'woocommerce-payments' ),
-		value: 31,
-	},
-];
+	...Array.from( { length: 28 }, ( _, i ) => i + 1 ),
+	31,
+].map( ( anchor ) => ( {
+	value: anchor,
+	label: getDepositMonthlyAnchorLabel( { monthlyAnchor: anchor } ),
+} ) );
```

Comment:
This is more concise and DRY (uses `getDepositMonthlyAnchorLabel` utility function) than the existing hard-coded labels.

Ive added a code comment above to add some context as it is harder to reason about at a glance.